packages/core/src/agent/index.ts

```diff
@@ -297,7 +297,6 @@ export class Agent<
               }
             : null,
           ...processedMessages,
-          ...newMessages,
```

Comment:
I found a bug while testing version `0.9.0-alpha.6`.

When I send a message to the agent, it doesn’t consider that message in the current response. It only uses it in the next interaction. This delay likely comes from removing `...newMessages`, which seems to skip the latest user input during processing.

Was this change intentional?

Reply:
We are taking a look! Thanks for pointing this out. With this in place we were duplicating messages. So something we will fix before releasing

Reply:
Incoming fix here! https://github.com/mastra-ai/mastra/pull/3763/files
Thanks for reporting it @andersonba 

I added a test too to make sure